### PR TITLE
feat: add elicitation confirmation and wallet name resolution

### LINKED ISSUE #73: Add elicitation for transaction confirmation before broadcast
## Context

The `sign_and_broadcast` and `sign_and_confirm` tools currently execute without any user confirmation. For real money operations, the MCP should ask "are you sure?" before broadcasting.

MCP elicitation allows the server to request structured input from the user mid-task — perfect for this.

## What to implement

Before broadcasting in `sign_and_broadcast` / `sign_and_confirm`, decode the transaction and present it for confirmation:

```
Server → Client (elicitation form):
┌─────────────────────────────────────┐
│ Confirm Transaction                 │
│                                     │
│ Type: TransferContract              │
│ From: TJD...  (wallet: "savings")   │
│ To:   TKX...                        │
│ Amount: 1,000 TRX                   │
│                                     │
│ Estimated fee: ~0.267 TRX           │
│                                     │
│ [Confirm]  [Cancel]                 │
└─────────────────────────────────────┘
```

If user cancels, return a tool result explaining the TX was not broadcast.

## Design

- Only elicit when client supports it (capability check)
- Skip elicitation for agent-controlled wallets (no restrictions in policy)
- Skip when `--require-policy` is not set (opt-in safety)
- Configurable threshold — only ask above X TRX amount
- Fall back to current behavior if elicitation unavailable

## How this replaces the crypto approval flow (#53)

For most use cases, elicitation is simpler than the full cryptographic approval flow:
- No wallet signature needed
- No intent hash computation
- No nonce tracking
- Built into MCP protocol — no custom infrastructure

The crypto approval flow (#53) is still valuable for remote/async approval (user on phone, agent running headless). But elicitation covers the common case of "agent and user in the same session."

## Value

Prevents accidental broadcasts. Makes the signing flow safe without complex infrastructure. Built on MCP protocol standard.

## Prerequisite

Requires mcp-go to support elicitation. Check current library support.

### PR DESCRIPTION
**Pull request by fbsobreira**
## Summary

- **Elicitation confirmation** before broadcasting transactions when no policy engine is active — the MCP server pauses and shows the user a confirmation dialog with decoded transaction details (type, from, to, amount)
- **Wallet name resolution** in `transfer_trx` and `transfer_trc20` tools — agents can now pass a wallet name in the `from` field instead of requiring raw base58 addresses
- **TRC20 human-readable amounts** in the confirmation dialog — resolves token decimals and symbol from the network

### How elicitation works

When `sign_and_broadcast` or `sign_and_confirm` is called without a policy engine, the server sends an elicitation request to the MCP client showing the decoded transaction details. The user clicks **Accept** to proceed or **Decline** to cancel.

- Policy-protected wallets skip elicitation (policy engine is the safety net)
- Headless clients (no elicitation support) proceed silently — same behavior as before
- No new config flags — client capability is the toggle

### Wallet name resolution

`transfer_trx` and `transfer_trc20` now accept wallet names in `from`:
```
transfer_trx(from: "savings", to: "TKSXDA8...", amount: "100")
```
Falls back to address validation if no wallet manager or name not found. Clear error message with hint to use `list_wallets`.

### Files changed

- `internal/server/server.go` — `WithElicitation()`, moved transfer registration after wallet manager init
- `internal/tools/sign.go` — `requestBroadcastConfirmation` helper, wired into both sign handlers
- `internal/tools/sign_test.go` — 13 new tests: mock sessions, full-flow accept/decline/not-supported/policy-skips
- `internal/tools/transfer.go` — `resolveFromAddress` helper, wallet name resolution
- `internal/tools/token.go` — wallet manager param for TRC20 transfer handler
- `internal/tools/transfer_test.go` — updated for new signatures
- `internal/wallet/manager.go` — exported `ResolveAddress`

Closes #73

## Test plan

- [x] `go test -race ./...` — all 16 packages pass
- [x] `golangci-lint run` — 0 issues
- [x] 4-agent code review (pr-review-toolkit, CodeRabbit, Codex MCP, Security Auditor)
- [ ] Manual: elicitation dialog shows for TRX transfer (no policy)
- [ ] Manual: user Accept → broadcasts successfully
- [ ] Manual: user Decline → cancelled, no broadcast
- [ ] Manual: with policy active → no dialog, broadcasts directly
- [ ] Manual: headless client → no dialog, broadcasts normally
- [ ] Manual: `transfer_trx(from: "wallet-name")` resolves correctly
- [ ] Manual: TRC20 confirmation shows human-readable amount + symbol

## Summary by CodeRabbit

* **New Features**
  * Interactive confirmation prompts before signing/broadcasting transactions, including token symbol and human-readable amount display.
  * Transfers accept wallet names as the sender ("from") in addition to raw addresses; sender resolution is attempted when a wallet is available.

* **Bug Fixes**
  * Cancelled confirmations now return a clear cancelled status and reason instead of proceeding.

### CODE REVIEW

File: internal/tools/sign.go [hunk 1]
```diff
@@ -529,6 +554,110 @@ func rebuildTransaction(ctx context.Context, pool *nodepool.Pool, intent *policy
...
+			// Try to resolve token decimals and symbol for human-readable display
+			if pool != nil {
+				token := trc20Std.New(pool.Client(), intent.TokenID)
+				if symbol, sErr := token.Symbol(ctx); sErr == nil {
+					if decimals, dErr := token.Decimals(ctx); dErr == nil && decimals > 0 {
+						divisor := math.Pow10(int(decimals))
+						humanAmount := intent.TokenAmount / divisor
+						fmt.Fprintf(&msg, "Amount: %g %s\n", humanAmount, symbol)
+					} else {
+						fmt.Fprintf(&msg, "Amount: %.0f raw units (%s)\n", intent.TokenAmount, symbol)
```
[the remark this answers is not in the document]
Comment (fbsobreira) [reply internal/tools/sign.go:620]: Fixed in ef18b9b. Replaced `math.Pow10` float64 division with `formatTokenAmount()` — inserts decimal point into `RawTokenAmt.String()` for exact precision. Falls back to `RawTokenAmt.String()` raw units when decimals unavailable. Added 8 table-driven tests covering nil, zero decimals, 6-decimal (USDT), 18-decimal tokens, and edge cases.

File: internal/tools/sign.go [hunk 2]
```diff
@@ -529,6 +554,110 @@ func rebuildTransaction(ctx context.Context, pool *nodepool.Pool, intent *policy
...
+			default:
+				fmt.Fprintf(&msg, "Amount: %v\n", amount)
+			}
+		}
+		if callValue, ok := decoded.Fields["call_value"]; ok {
+			// SDK converts call_value via sunToTRX() — returns string
+			if cv, isStr := callValue.(string); isStr && cv != "" && cv != "0.000000" {
+				fmt.Fprintf(&msg, "Call Value: %s TRX\n", cv)
+			}
+		}
```
[the remark this answers is not in the document]
Comment (fbsobreira) [reply internal/tools/sign.go:662]: The fallback path displays whatever fields the SDK decodes for a given contract type. Contracts like DelegateResource, FreezeBalance etc. don't typically flow through sign_and_broadcast in normal agent workflows — they use dedicated tools. Adding normalization for every possible contract type would be over-engineering for a confirmation message. The current behavior is honest: it shows what's decoded, and the Type field always appears so the user knows what they're confirming.

File: internal/tools/transfer_test.go [hunk 3]
```diff
@@ -15,7 +15,7 @@ import (
 
 func TestTransferTRX_InvalidFromAddress(t *testing.T) {
 	pool := newMockPool(t, &mockWalletServer{})
-	result := callTool(t, handleTransferTRX(pool), map[string]any{
+	result := callTool(t, handleTransferTRX(pool, nil), map[string]any{
```
[the remark this answers is not in the document]
Comment (fbsobreira) [reply internal/tools/transfer_test.go:19]: Already addressed — the amended commit includes `TestResolveFromAddress_WithWM_WalletName`, `TestResolveFromAddress_WithWM_RawAddress`, `TestResolveFromAddress_WithWM_UnknownName`, `TestResolveFromAddress_NilWM_ValidAddress`, and `TestResolveFromAddress_NilWM_InvalidAddress` covering all branches including the `list_wallets` hint.

File: internal/tools/transfer.go [hunk 4]
```diff
@@ -7,51 +7,78 @@ import (
...
+			return addr, nil
+		}
+	}
+	if err := validateAddress(nameOrAddr); err != nil {
+		if wm != nil {
+			return "", fmt.Errorf("invalid from: %q is not a valid TRON address or known wallet name — use list_wallets to find the address", nameOrAddr)
+		}
+		return "", fmt.Errorf("invalid from address: %v", err)
+	}
+	return nameOrAddr, nil
```
[the remark this answers is not in the document]
Comment (fbsobreira) [reply internal/tools/transfer.go:35]: Fixed in ef18b9b. Added `validateAddress(addr)` check after `ResolveAddress` succeeds, so malformed 34-char T-strings that pass the heuristic are caught before reaching the SDK.
